core/trino-main/src/main/java/io/trino/server/ThrowableMapper.java

```diff
@@ -45,20 +51,53 @@ public ThrowableMapper(ServerConfig config)
     @Override
     public Response toResponse(Throwable throwable)
     {
-        if (throwable instanceof WebApplicationException) {
-            return ((WebApplicationException) throwable).getResponse();
-        }
+        return switch (throwable) {
```

Comment:
This looks like step backward - can you explain inline why we need to have such an elaborate switch statemement here? Can we be less verbose?

Reply:
We can move it to Airlift as this isn't Trino specific.

Reply:
Previously error messages were generated in two layers: sometimes by jax-rs, sometimes by Jetty (setError was used for that). setError isn't the right approach with HTTP/2 as in the HTTP/2 there is no status line anymore. If you want to emit and error that contains a particular error line you need to use body for that.

Reply:
After this change, the error mapping to a response happens in a single place with removes disambiguation and makes it explicit.